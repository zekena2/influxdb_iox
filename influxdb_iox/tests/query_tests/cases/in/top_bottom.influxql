-- IOX_SETUP: top_bottom

--
-- top
--
SELECT top(writes, 2) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001;
SELECT top(usage_system,3) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001 AND cpu = 'cpu0';
SELECT top(usage_idle,5), cpu FROM cpu GROUP BY machine;
SELECT top(writes,3) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001 GROUP BY time(7s);
SELECT top(writes,2) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001 GROUP BY time(30s);
SELECT top(usage_system,machine,cpu,2) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
SELECT top(usage_system,machine,2),cpu FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
SELECT top(usage_system,machine,2),machine FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
SELECT top(usage_idle,machine,cpu,2) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001 GROUP BY TIME(60s);
--
-- bottom
--
SELECT bottom(reads, 3) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001;
SELECT bottom(usage_system,3) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001 AND cpu = 'cpu1';
SELECT bottom(usage_idle,5), cpu FROM cpu GROUP BY machine;
SELECT bottom(writes,3) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001 GROUP BY time(7s);
SELECT bottom(writes,2) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001 GROUP BY time(30s);
SELECT bottom(usage_system,machine,cpu,2) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
SELECT bottom(usage_system,machine,2),cpu FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
SELECT bottom(usage_system,machine,2),machine FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
SELECT bottom(usage_idle,machine,cpu,2) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001 GROUP BY TIME(60s);