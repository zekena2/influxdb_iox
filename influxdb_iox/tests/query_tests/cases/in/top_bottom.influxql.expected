-- Test Setup: top_bottom
-- InfluxQL: SELECT top(writes, 2) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001;
name: diskio
+---------------------+---------+
| time                | top     |
+---------------------+---------+
| 1970-01-01T00:03:20 | 5593589 |
| 1970-01-01T00:03:30 | 5593735 |
+---------------------+---------+
-- InfluxQL: SELECT top(usage_system,3) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001 AND cpu = 'cpu0';
name: cpu
+---------------------+------+
| time                | top  |
+---------------------+------+
| 1970-01-01T00:02:10 | 89.8 |
| 1970-01-01T00:02:50 | 89.8 |
| 1970-01-01T00:03:00 | 90.0 |
+---------------------+------+
-- InfluxQL: SELECT top(usage_idle,5), cpu FROM cpu GROUP BY machine;
name: cpu
tags: machine=machine1
+---------------------+------+------+
| time                | top  | cpu  |
+---------------------+------+------+
| 1970-01-01T00:01:00 | 99.8 | cpu1 |
| 1970-01-01T00:01:20 | 99.8 | cpu1 |
| 1970-01-01T00:02:00 | 99.9 | cpu1 |
| 1970-01-01T00:02:20 | 99.9 | cpu1 |
| 1970-01-01T00:02:30 | 99.9 | cpu1 |
+---------------------+------+------+
name: cpu
tags: machine=machine2
+---------------------+------+------+
| time                | top  | cpu  |
+---------------------+------+------+
| 1970-01-01T00:01:00 | 89.8 | cpu1 |
| 1970-01-01T00:01:20 | 89.8 | cpu1 |
| 1970-01-01T00:02:00 | 89.9 | cpu1 |
| 1970-01-01T00:02:20 | 89.9 | cpu1 |
| 1970-01-01T00:02:30 | 89.9 | cpu1 |
+---------------------+------+------+
-- InfluxQL: SELECT top(writes,3) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001 GROUP BY time(7s);
name: diskio
+---------------------+---------+
| time                | top     |
+---------------------+---------+
| 1970-01-01T00:02:10 | 5592646 |
| 1970-01-01T00:02:20 | 5592810 |
| 1970-01-01T00:02:30 | 5592997 |
| 1970-01-01T00:02:40 | 5593109 |
| 1970-01-01T00:02:50 | 5593219 |
| 1970-01-01T00:03:00 | 5593438 |
| 1970-01-01T00:03:10 | 5593513 |
| 1970-01-01T00:03:20 | 5593589 |
| 1970-01-01T00:03:30 | 5593735 |
+---------------------+---------+
-- InfluxQL: SELECT top(writes,2) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001 GROUP BY time(30s);
name: diskio
+---------------------+---------+
| time                | top     |
+---------------------+---------+
| 1970-01-01T00:02:10 | 5592646 |
| 1970-01-01T00:02:20 | 5592810 |
| 1970-01-01T00:02:40 | 5593109 |
| 1970-01-01T00:02:50 | 5593219 |
| 1970-01-01T00:03:10 | 5593513 |
| 1970-01-01T00:03:20 | 5593589 |
| 1970-01-01T00:03:30 | 5593735 |
+---------------------+---------+
-- InfluxQL: SELECT top(usage_system,machine,cpu,2) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
name: cpu
+---------------------+------+----------+------+
| time                | top  | machine  | cpu  |
+---------------------+------+----------+------+
| 1970-01-01T00:02:00 | 99.9 | machine1 | cpu1 |
| 1970-01-01T00:03:00 | 90.0 | machine1 | cpu0 |
+---------------------+------+----------+------+
-- InfluxQL: SELECT top(usage_system,machine,2),cpu FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
name: cpu
+---------------------+------+----------+------+
| time                | top  | machine  | cpu  |
+---------------------+------+----------+------+
| 1970-01-01T00:02:00 | 99.9 | machine1 | cpu1 |
| 1970-01-01T00:02:00 | 89.9 | machine2 | cpu1 |
+---------------------+------+----------+------+
-- InfluxQL: SELECT top(usage_system,machine,2),machine FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
name: cpu
+---------------------+------+----------+-----------+
| time                | top  | machine  | machine_1 |
+---------------------+------+----------+-----------+
| 1970-01-01T00:02:00 | 99.9 | machine1 | machine1  |
| 1970-01-01T00:02:00 | 89.9 | machine2 | machine2  |
+---------------------+------+----------+-----------+
-- InfluxQL: SELECT top(usage_idle,machine,cpu,2) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001 GROUP BY TIME(60s);
name: cpu
+---------------------+------+----------+------+
| time                | top  | machine  | cpu  |
+---------------------+------+----------+------+
| 1970-01-01T00:01:00 | 99.8 | machine1 | cpu1 |
| 1970-01-01T00:01:00 | 89.8 | machine2 | cpu1 |
| 1970-01-01T00:02:00 | 99.9 | machine1 | cpu1 |
| 1970-01-01T00:02:30 | 90.4 | machine1 | cpu0 |
| 1970-01-01T00:03:00 | 99.8 | machine1 | cpu1 |
| 1970-01-01T00:03:00 | 90.0 | machine1 | cpu0 |
+---------------------+------+----------+------+
-- InfluxQL: SELECT bottom(reads, 3) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001;
name: diskio
+---------------------+---------+
| time                | bottom  |
+---------------------+---------+
| 1970-01-01T00:02:10 | 2592646 |
| 1970-01-01T00:02:30 | 2592997 |
| 1970-01-01T00:02:50 | 2593219 |
+---------------------+---------+
-- InfluxQL: SELECT bottom(usage_system,3) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001 AND cpu = 'cpu1';
name: cpu
+---------------------+--------+
| time                | bottom |
+---------------------+--------+
| 1970-01-01T00:01:00 | 89.8   |
| 1970-01-01T00:01:10 | 89.7   |
| 1970-01-01T00:02:10 | 89.8   |
+---------------------+--------+
-- InfluxQL: SELECT bottom(usage_idle,5), cpu FROM cpu GROUP BY machine;
name: cpu
tags: machine=machine1
+---------------------+--------+------+
| time                | bottom | cpu  |
+---------------------+--------+------+
| 1970-01-01T00:01:10 | 88.6   | cpu0 |
| 1970-01-01T00:01:20 | 88.6   | cpu0 |
| 1970-01-01T00:01:30 | 83.4   | cpu0 |
| 1970-01-01T00:01:40 | 87.7   | cpu0 |
| 1970-01-01T00:02:00 | 86.9   | cpu0 |
+---------------------+--------+------+
name: cpu
tags: machine=machine2
+---------------------+--------+------+
| time                | bottom | cpu  |
+---------------------+--------+------+
| 1970-01-01T00:01:10 | 78.6   | cpu0 |
| 1970-01-01T00:01:20 | 78.6   | cpu0 |
| 1970-01-01T00:01:30 | 73.4   | cpu0 |
| 1970-01-01T00:01:40 | 77.7   | cpu0 |
| 1970-01-01T00:02:00 | 76.9   | cpu0 |
+---------------------+--------+------+
-- InfluxQL: SELECT bottom(writes,3) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001 GROUP BY time(7s);
name: diskio
+---------------------+---------+
| time                | bottom  |
+---------------------+---------+
| 1970-01-01T00:02:10 | 5592646 |
| 1970-01-01T00:02:20 | 5592810 |
| 1970-01-01T00:02:30 | 5592997 |
| 1970-01-01T00:02:40 | 5593109 |
| 1970-01-01T00:02:50 | 5593219 |
| 1970-01-01T00:03:00 | 5593438 |
| 1970-01-01T00:03:10 | 5593513 |
| 1970-01-01T00:03:20 | 5593589 |
| 1970-01-01T00:03:30 | 5593735 |
+---------------------+---------+
-- InfluxQL: SELECT bottom(writes,2) FROM diskio WHERE time >= 0000000130000000000 AND time < 0000000210000000001 GROUP BY time(30s);
name: diskio
+---------------------+---------+
| time                | bottom  |
+---------------------+---------+
| 1970-01-01T00:02:10 | 5592646 |
| 1970-01-01T00:02:20 | 5592810 |
| 1970-01-01T00:02:30 | 5592997 |
| 1970-01-01T00:02:40 | 5593109 |
| 1970-01-01T00:03:00 | 5593438 |
| 1970-01-01T00:03:10 | 5593513 |
| 1970-01-01T00:03:30 | 5593735 |
+---------------------+---------+
-- InfluxQL: SELECT bottom(usage_system,machine,cpu,2) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
name: cpu
+---------------------+--------+----------+------+
| time                | bottom | machine  | cpu  |
+---------------------+--------+----------+------+
| 1970-01-01T00:01:30 | 73.4   | machine2 | cpu0 |
| 1970-01-01T00:01:30 | 83.4   | machine1 | cpu0 |
+---------------------+--------+----------+------+
-- InfluxQL: SELECT bottom(usage_system,machine,2),cpu FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
name: cpu
+---------------------+--------+----------+------+
| time                | bottom | machine  | cpu  |
+---------------------+--------+----------+------+
| 1970-01-01T00:01:30 | 73.4   | machine2 | cpu0 |
| 1970-01-01T00:01:30 | 83.4   | machine1 | cpu0 |
+---------------------+--------+----------+------+
-- InfluxQL: SELECT bottom(usage_system,machine,2),machine FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001;
name: cpu
+---------------------+--------+----------+-----------+
| time                | bottom | machine  | machine_1 |
+---------------------+--------+----------+-----------+
| 1970-01-01T00:01:30 | 83.4   | machine1 | machine1  |
| 1970-01-01T00:01:30 | 73.4   | machine2 | machine2  |
+---------------------+--------+----------+-----------+
-- InfluxQL: SELECT bottom(usage_idle,machine,cpu,2) FROM cpu WHERE time >= 0000000060000000000 AND time < 0000000210000000001 GROUP BY TIME(60s);
name: cpu
+---------------------+--------+----------+------+
| time                | bottom | machine  | cpu  |
+---------------------+--------+----------+------+
| 1970-01-01T00:01:30 | 73.4   | machine2 | cpu0 |
| 1970-01-01T00:01:30 | 83.4   | machine1 | cpu0 |
| 1970-01-01T00:02:00 | 76.9   | machine2 | cpu0 |
| 1970-01-01T00:02:00 | 86.9   | machine1 | cpu0 |
| 1970-01-01T00:03:10 | 78.8   | machine2 | cpu0 |
| 1970-01-01T00:03:10 | 88.8   | machine1 | cpu0 |
+---------------------+--------+----------+------+